### ISSUE TITLE
Template files instead of inline ones?

### ISSUE THREAD

**Issue by aolko** [cferdinandi/reef#127]
Why not just define the tags in a separate [template] file, instead of shoving html inline?
see: [vue](https://vuejs.org/v2/guide/single-file-components.html), [riot](https://riot.js.org/#:~:text=Riot.js%20brings%20custom%20elements%20to%20all%20modern%20browsers%20without%20the%20use%20of%20any%20polyfill!)

**Comment by cferdinandi** [comment]
I don't understand the question. Can you rephrase?

**Comment by aolko** [comment]
> I don't understand the question. Can you rephrase?

Why not instead of inline template definitions
```
let app = new Reef('#app', {
	template: '<h1>Hello, world!</h1>'
});
```
we would load the template file with the template, styles and scripts already in place 
![](https://aolko.is-inside.me/CGdadhe3.png)
and just mount it?

**Comment by cferdinandi** [comment]
So...

1. Reef doesn't use JSX
2. If that's the kind of experience you want, feel free to use a different tool

That's a bit like asking why Vue does it the way you described instead of how Reef handles things. Different tools, different approaches. 😎

**Comment by aolko** [comment]
> 2\. If that's the kind of experience you want, feel free to use a different tool

except there's very little if none different tools, being small js libraries that do just that and allow to ignore w3c cusom elements standard

**Comment by cferdinandi** [comment]
Two that immediately come to mind:

1. AlpineJS: https://alpinejs.dev/
2. Petite Vue, from Vue creator Evan You: https://twitter.com/youyuxi/status/1410763549228609537

**Comment by aolko** [comment]
> 1. AlpineJS: https://alpinejs.dev/

alpine doesn't quite cut it, it sits among the likes of htmx/intercooler, [trimmings](https://github.com/postlight/trimmings) and alike, those are heavily reliant on dom and do not have the file templating and/or injection of "non-standard tags"



> 2\. Petite Vue, from Vue creator Evan You: https://twitter.com/youyuxi/status/1410763549228609537

and petite vue's templating is kinda weak, it assumes that it has to be on the same page, not mounted from a separate file

**Comment by cferdinandi** [comment]
Sounds like you want a build tool of sorts. Svelte maybe? Either way, Reef is very much not trying to be that. 

Or, since you have a strong vision around this, you could fork and roll your own?

**Comment by aolko** [comment]
> Sounds like you want a build tool of sorts. Svelte maybe? Either way, Reef is very much not trying to be that.

no i don't, i want a runtime library




> Or, since you have a strong vision around this, you could fork and roll your own?

sadly it isn't written in es5 (in only which i write) and i have no clue on how to implement that
